pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/db/dto/EventConfigurationDto.java

```diff
@@ -13,332 +16,106 @@
 import org.broadinstitute.ddp.model.event.NotificationType;
 import org.broadinstitute.ddp.model.event.PdfAttachment;
 import org.broadinstitute.ddp.model.user.EnrollmentStatusType;
+import org.jdbi.v3.core.mapper.reflect.ColumnName;
+import org.jdbi.v3.core.mapper.reflect.JdbiConstructor;
 
 /**
  * A DTO representing the event configuration, left joined with possible trigger and action configuration info
  */
+@Value
+@AllArgsConstructor(onConstructor = @__(@JdbiConstructor))
```

Comment:
This is listed as an experimental feature in Lombok's [docs](https://projectlombok.org/features/experimental/onX). I'm hesitant about relying on this feature. What are your thoughts, @MocanaAtBroad

Reply:
Thanks for flagging this, @bskinner . I am inclined to go ahead and use it anyways because it really is a very useful option.
Just thinking that one of 3 things could happen:
1) Capability remains as is (in this case we are golden)
2) Capability is modified changing the annotation name/syntax. (In this case we could do a global replace).
3) Capability is totally dropped from Lombok (In this case we would have to add back all the constructor stuff).

1 or 2 seem to A LOT more probable than 3 , so would be no op or minor refactor. And 3 would not be the end of the world.
I say let's go for it. I am just miffed that we are up to Java 9001 and we still need to write all this mindless boilerplate.

Reply:
> Thanks for flagging this, @bskinner . I am inclined to go ahead and use it anyways because it really is a very useful option.

Agreed with it being a useful feature- I'm all for reducing the amount of boilerplate we're writing.

>  I am just miffed that we are up to Java 9001 and we still need to write all this mindless boilerplate.

No kidding- It'd be lovely when we can finally drop this. When Objective-C introduced the `@synthesize` keyword and Automatic Property Synthesis, it was a game changer.

@Trimax :
Walking through the docs again, I ran across this blurb:
```
The syntax is a little strange and depends on the javac you are using.
On javac7, to use any of the 3 onX features, you must wrap the annotations to be applied to the constructor / method / parameter in @__(@AnnotationGoesHere). To apply multiple annotations, use @__({@Annotation1, @Annotation2}). The annotations can themselves obviously have parameters as well.
On javac8 and up, you add an underscore after onMethod, onParam, or onConstructor. 
```

It _seems_ like things are working, but do we need to change things here, or is this an outdated note? The use of `onConstructor` without an underscore seems to be counter to what the docs state given we're on Java 11.

Reply:
I tested it even with Java 17 and it is working as expected. Besides, no one canceled backward compatibility :)

Reply:
I'm not sure where backward compatibility comes into this?